src/components/app/details/cdDetails/CDDetails.tsx

```diff
@@ -131,7 +131,7 @@ export default function CDDetails() {
 
     useEffect(() => {
         if (result) {
-            const pipelines = result[1]['value']?.pipelines
+            const pipelines = result[1]['value']?.pipelines || []
             const _deploymentAppType = pipelines?.find(
```

Comment:
Will it affect further as it is undefined for empty

Reply:
The only case which i believe could be causing an issue is if envData.environmentId is not in cdPipelinesMap, which I believe wont happen as in that case result from api wont be there, but I have also added an optional check in case.